### PR TITLE
Add emplace_back to edm::VecArray

### PR DESCRIPTION
**Pull request by makortel**
This PR adds `emplace_back` to `edm::VecArray` mostly for syntactic sugar for the case that `VecArray` holds e.g. `std::pair` (i.e. one can avoid typing `std::make_pair(...)` or `std::pair<...>(...)` that are needed with the existing `push_back`). It should nevertheless be noted that this is not a "true" `emplace_back` like in `std::vector` since all elements of `VecArray::data_` are value-initialized in the constructor of `VecArray`. On the other hand, `VecArray` makes sense only for types that are cheap to construct and copy, so in practice there should be no (added) problems.

Tested in 9_4_0_pre3, no changes expected.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: FWCore/Utilities/test/vecarray.cppunit.cpp [hunk 1]
```diff
@@ -63,6 +63,13 @@ void testVecArray::test() {
     CPPUNIT_ASSERT(true);
   }
 
+  try {
+    array.emplace_back(5);
```
Comment (Dr15Jones) [inline FWCore/Utilities/test/vecarray.cppunit.cpp]: CPPUNIT_EXCEPTION is simpler

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
+1

**Comment by davidlange6** [comment]
merge

**Pull request merged**